service/matching/reachability.go

```diff
@@ -73,14 +93,26 @@ func getBuildIdTaskReachability(
 		redirectRules:                data.GetRedirectRules(),
 		buildIdVisibilityGracePeriod: buildIdVisibilityGracePeriod,
 	}
+}
 
-	return rc.run(ctx, buildId)
+func getBuildIdTaskReachability(
+	ctx context.Context,
+	rc *reachabilityCalculator,
+	metricsHandler metrics.Handler,
+	buildId string,
+) (enumspb.BuildIdTaskReachability, error) {
+	reachability, exitPoint, err := rc.run(ctx, buildId)
+	metrics.ReachabilityExitPointCounter.With(metricsHandler.WithTags(
+		metrics.NamespaceTag(rc.nsName.String()),
+		metrics.TaskQueueTag(rc.taskQueue)),
+	).Record(1, metrics.StringTag(reachabilityExitPointTagName, reachabilityExitPoint2TagValue[exitPoint]))
+	return reachability, err
```

Comment:
let's also log the reachability result and step so if a customer complains about wrong reachability result we can have the logs to investigate.